schema/platform/archetype/location/create.ts

```diff
@@ -77,13 +77,81 @@ export async function createLocation(
+    // worktemplateconstraint and worktemplatenexttemplate. This is sort of a
+    // hack, but in reality this should be built into the next version of the
+    // rules engine. Why worktemplatenexttemplate? Just because a template *can*
+    // be instantiated at a location does not necessarily mean that we should
+    // instantiate it. This is where eager vs lazy instantiation comes into
+    // play. The eager case tells us that we *should* instantiate, while the
+    // lazy case tells us to hold off. If we do *not* listen we will end up over
+    // instantiating in some cases, e.g. under Runtime we would instantiate Run,
+    // Idle Time and Downtime which is obviously not desired.
+    {
+      const result = await sql`
+        with to_instantiate as (
+            select
+                wt.id as template_id,
+                loc.locationuuid as location_id,
+                auth.current_identity(loc.locationcustomerid, ${ctx.auth.userId}) as modified_by
+            from public.location as loc
+            inner join public.worktemplate as wt
+                -- Ensure that we only evaluate templates that are in scope
+                on loc.locationsiteid = wt.worktemplatesiteid
+            inner join public.worktemplatenexttemplate as nt
+                -- Only templates that use "eager" instantiation
+                on  wt.worktemplateid = nt.worktemplatenexttemplateprevioustemplateid
+                and wt.worktemplateid = nt.worktemplatenexttemplatenexttemplateid
+                and (
+                    nt.worktemplatenexttemplateenddate is null
+                    or nt.worktemplatenexttemplateenddate > now()
+                )
+            where
+                loc.locationuuid = ${row.id}
+                -- Only templates that *can* be instantiated at the given location *type*
+                and exists (
+                    select 1
+                    from public.worktemplateconstraint as wtc
+                    where
+                        wtc.worktemplateconstrainttemplateid = wt.id
+                        and wtc.worktemplateconstraintconstraintid = (
+                            select custaguuid
+                            from public.custag
+                            where custagid = loc.locationcategoryid
+                        )
+                        and wtc.worktemplateconstraintconstrainedtypeid = (
+                            select systaguuid
+                            from public.systag
+                            where systagparentid = 849 and systagtype = 'Location'
+                        )
+                    limit 1
+                )
+        )
+
+        select 1
+        from
+            to_instantiate as t,
+            engine0.instantiate(
+                template_id := t.template_id,
+                location_id := t.location_id,
+                target_state := 'Open',
+                target_type := 'On Demand',
```

Comment:
This is likely incorrect? We should probably stick to using worktemplatenexttemplatetypeid here, per existing engine0 logic. This could, in theory, cause instantiations with "incorrect" work types if one of the rules indicates a task is of type Task (rather than On Demand). In fact, we likely need to limit this to _just_ rules targeting Tasks and On Demands, so we don't inadvertently instantiate Audits and/or Remediations :P